pidgin/app.py

```diff
@@ -68,6 +83,26 @@ def flatten_dict(d):
         raise NoCoreMetadataException(error)
     return flat_d
 
+def generate_ciation(metadata_dict):
+    """
+    Generate a citation from the other metadata.
+    """
+    creator = metadata_dict.get('creator')
+    year = metadata_dict.get('updated_datetime').split('-')[0]
+    title = metadata_dict.get('title')
+    publisher = metadata_dict.get('publisher')
+    guid = metadata_dict.get('object_id')
+    return '{}, {}: {}. {}, {}.'.format(creator, year, title, publisher, guid)
```

Comment:
There's a fun trick with `format` accepting keyword arguments you could use here:
```python
format_args = dict(metadata_dict)
format_args['year'] = format_args.pop('updated_datetime').split('-')[0]
format_string = '{creator}, {year}: {title}. {publisher}, {object_id}'
return format_string.format(**format_args)
```